src/ansys/additive/core/parametric_study/parametric_study.py

```diff
@@ -1210,55 +1218,234 @@ def add_inputs(
+            ]
+
+            if overwrite:
+                # Drop duplicates and keep the latest completed simulation entry in case of adding a
+                # completed simulation when using add_summaries.
+                # Simulation status further narrows down subset of columns to check.
+                single_bead_df.drop_duplicates(
+                    subset=sb_params + [ColumnNames.STATUS],
+                    ignore_index=True,
+                    keep="last",
+                    inplace=True,
+                )
+                porosity_df.drop_duplicates(
+                    subset=porosity_params + [ColumnNames.STATUS],
+                    ignore_index=True,
+                    keep="last",
+                    inplace=True,
+                )
+                microstructure_df.drop_duplicates(
+                    subset=microstructure_params + [ColumnNames.STATUS],
+                    ignore_index=True,
+                    keep="last",
+                    inplace=True,
+                )
+
+            # Drop duplicates and keep the earlier entry in case of adding a pending/skip simulation
+            # when using add_inputs.
+            # Completed simulations will remain as is since they are already sorted and are higher
+            # up in the list.
+            if len(single_bead_df) > 0:
+                duplicates_removed_df = pd.concat(
+                    [
+                        duplicates_removed_df,
+                        single_bead_df.drop_duplicates(
+                            subset=sb_params, ignore_index=True, keep="first"
+                        ),
+                    ]
+                )
+            if len(porosity_df) > 0:
+                duplicates_removed_df = pd.concat(
+                    [
+                        duplicates_removed_df,
+                        porosity_df.drop_duplicates(
+                            subset=porosity_params, ignore_index=True, keep="first"
+                        ),
+                    ]
+                )
+            if len(microstructure_df) > 0:
+                duplicates_removed_df = pd.concat(
+                    [
+                        duplicates_removed_df,
+                        microstructure_df.drop_duplicates(
+                            subset=microstructure_params, ignore_index=True, keep="first"
+                        ),
+                    ]
+                )
+            if duplicates_removed_df.__len__() < current_df.__len__():
+                self._data_frame = duplicates_removed_df
+                LOG.debug(
+                    f"Removed {current_df.__len__()-duplicates_removed_df.__len__()} duplicate simulation(s)."
+                )
+            else:
+                LOG.debug("No duplicate simulations found.")
+            return current_df.__len__() - duplicates_removed_df.__len__()
+        except Exception as e:
+            LOG.error(f"Error removing duplicate simulations: {e}")
```

Comment:
I could be wrong, but I don't think we want to just log an error here. I think we want to stop what we're doing by either re-raising the exception or not using a try block in the first place. My personal preference would be not to use a try block.

In general, I only use a try block if I can do something after catching the exception, like cleaning up a resource, or if I want to make sure code gets executed in a `finally` case. Usually, I let a higher stack level handle the exception.

Reply:
I agree, we do not require a `try-catch` block if there is no clean up.